src/extensions/scratch3_text2speech/index.js

```diff
@@ -584,14 +595,44 @@ class Scratch3Text2SpeechBlocks {
     }
 
     /**
-     * Get the menu of languages for the "set language" block.
+     * Get the localized menu of languages for the "set language" block.
+     * For each language:
+     *   if there is a custom translated spoken language name, use that;
+     *   otherwise use the translation in the languageNames menuMap;
+     *   otherwise fall back to the untranslated name in LANGUAGE_INFO.
      * @return {array} the text and value for each menu item.
      */
     getLanguageMenu () {
-        return Object.keys(this.LANGUAGE_INFO).map(key => ({
-            text: this.LANGUAGE_INFO[key].name,
-            value: key
-        }));
+        const editorLanguage = this.getEditorLanguage();
+        // Get the array of localized language names
+        const localizedNameMap = {};
+        let nameArray = languageNames.menuMap[editorLanguage];
+        if (nameArray) {
+            // Also get any localized names of spoken languages
+            let spokenNameArray = [];
+            if (languageNames.spokenLanguages) {
+                spokenNameArray = languageNames.spokenLanguages[editorLanguage];
+                nameArray = nameArray.concat(spokenNameArray);
+            }
+            // Create a map of language code to localized name
+            nameArray.forEach(lang => {
```

Comment:
Maybe add a note about how we don't end up with duplicates for Chinese and Mandarin in the array.

Reply:
And a test for it :)

Reply:
all set!